kyo-stm/shared/src/main/scala/kyo/TRefLog.scala

```diff
@@ -2,25 +2,23 @@ package kyo
 
 /** A log of transactional operations performed on TRefs within an STM transaction.
   *
-  * RefLog maintains a mapping from transactional references to their pending read/write operations within a transaction. It tracks both
+  * TRefLog maintains a mapping from transactional references to their pending read/write operations within a transaction. It tracks both
   * read entries (which record the version of data read) and write entries (which contain the new values to be committed).
   *
   * This type is used internally by the STM implementation and should not be accessed directly by application code.
   *
   * @note
   *   This is a private implementation detail of the STM system
   */
-opaque type RefLog = Map[TRef[Any], RefLog.Entry[Any]]
+opaque type TRefLog = Map[TRef[Any], TRefLog.Entry[Any]]
 
-private[kyo] object RefLog:
+private[kyo] object TRefLog:
 
-    given Tag[RefLog] = Tag[Map[TRef[Any], Entry[Any]]]
```

Comment:
There was an issue with the handling of `Var[RefLog]` producing a dynamic tag with the `RefLog` type bound at runtime. I've moved all methods that use the tag to the companion object so the tag can be fully inferred statically. I've also played with replacing `Map` with `TreeMap` but the results weren't encouraging.